src/main/java/com/gregtechceu/gtceu/api/block/MetaMachineBlock.java

```diff
@@ -263,6 +265,10 @@ public InteractionResult use(BlockState state, Level world, BlockPos pos, Player
             if (result.getSecond() != InteractionResult.PASS) return result.getSecond();
         }
 
+        if (itemStack.is(GTItems.PORTABLE_SCANNER.get())) {
```

Comment:
This was the best way I found to prevent rightclick from opening the UI and instead scan the machine. Let me know if there is a better way